app/src/ui/multi-commit-operation/squash.tsx

```diff
@@ -0,0 +1,79 @@
+import { RebaseConflictState } from '../../lib/app-state'
+import { MultiCommitOperationKind } from '../../models/multi-commit-operation'
+import { MultiCommitOperation } from './multi-commit-operation'
+
+export abstract class Squash extends MultiCommitOperation {
+  protected onBeginOperation = () => {
+    const { repository, dispatcher, state } = this.props
+    const { commits, operationDetail } = state
+
+    if (operationDetail.kind !== MultiCommitOperationKind.Squash) {
+      this.endFlowInvalidState()
+      return
+    }
+
+    const {
+      targetCommit,
+      lastRetainedCommitRef,
+      commitContext,
+    } = operationDetail
+
+    return dispatcher.squash(
+      repository,
+      commits,
+      targetCommit,
+      lastRetainedCommitRef,
+      commitContext
+    )
+  }
+
+  protected onContinueAfterConflicts = async (): Promise<void> => {
+    const {
+      repository,
+      dispatcher,
+      workingDirectory,
+      state,
+      conflictState,
+    } = this.props
+    const { commits, currentTip, targetBranch, originalBranchTip } = state
+
+    if (conflictState === null) {
+      this.endFlowInvalidState()
+      return
+    }
+
+    await dispatcher.switchMultiCommitOperationToShowProgress(repository)
+
+    const rebaseConflictState: RebaseConflictState = {
```

Comment:
This is another one of those places that if I refactor the way the conflict state works this section will become simpler in not needing to build out a rebase conflict state... but for a later time and probably not time prudent to focus on before other things (squash force push, merge squash, some squash error handling)